File: hw/tipantiza/u3/hw24AbstractFactory/AbstractFactory/src/main/java/ec/edu/espe/abstractfactory/model/GUIFactory.java

package ec.edu.espe.abstractfactory.model;
import ec.edu.espe.abstractfactory.controller.ConfigReader;
/**
 *
 * @author Alexander Tipantiza, The Softwarrios, @ESPE
 */

public abstract class GUIFactory {
    
    public static GUIFactory getFactory() {
        int sys = ConfigReader.getOSType();
        if (sys == 0) {
            return new WinFactory();
        } else if (sys == 2) {
            return new MacFactory();
        } else {
            return new LinuxFactory();
        }
    }
    
    public abstract Button createButton();
    public abstract Menu createMenu();
}

File: hw/tipantiza/u3/hw24AbstractFactory/AbstractFactory_py/model/gui_factory.py
from abc import ABC, abstractmethod
from controller.config_reader import ConfigReader

class GUIFactory(ABC):
    
    @staticmethod
    def get_factory():
        sys_type = ConfigReader.get_os_type()
        if sys_type == 0:
            from model.win_factory import WinFactory
            return WinFactory()
        elif sys_type == 2:
            from model.mac_factory import MacFactory
            return MacFactory()
        else:
            from model.linux_factory import LinuxFactory
            return LinuxFactory()
    
    @abstractmethod
    def create_button(self):
        pass
    
    @abstractmethod
    def create_menu(self):
        pass